### PR TITLE
chore(releasing): Add documentation about Vector's release process

### PR DESCRIPTION
**Pull request by jszwedko**
Addresses #6733 

I was going to write an RFC, but then just figured I'd write a document to publish about it that we can all align around.

Particular issues I'm looking to address with this:

* Blocking releases while we wait for large features. In the past we did this to create more interesting releases, but I think now a more regular cadence is preferable to set user expectations and ensure we don't pile up too many changes at once, increasing the potential for regressions. This does mean some releases will be more interesting than others. Some might just contain small enhancements and bug fixes while others contain features like VRL. I personally think this is fine but also that there will be useful things to highlight in each release given a 6 week cadence.
* Giving users some expectation around cadence
* Giving users some expectation around compatibility
* Letting users know how to keep abreast of new releases

We can flesh this out in the future with improvements like release candidates and betas but I think this is a good starting point to iterate on.

Inspired by https://github.com/microsoft/TypeScript-wiki/blob/master/TypeScript's-Release-Process.md

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: docs/manual/about/release-process.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+We aim to keep Vector as backwards compatible as possible, preferring
+deprecation over breaking compatibly, but we will occasionally introduce
+backwards incompatible changes as we learn better ways of doing things .
+
+While Vector is pre-1.0, we will only make backwards incompatible changes in
+minor releases (for example from `0.11.1` to `0.12.0`). We will not make
+backwards incompatible changes in point releases (for example from `0.11.0` to
+`0.11.1`) unless there is a critical bug that must be addressed that requires
+breaking compatibility to fix (this has never happened yet).
```
Comment (jszwedko) [inline docs/manual/about/release-process.md:76]: At least to my knowledge 😅

File: docs/manual/about/release-process.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+---
+
+Here you can find information about how to track Vector's releases, how often
+we release, compatibility guarantees, and more.
+
+## How to track releases
+
+There are a few different ways to track when there is a new release:
+
+* Following [@vectordotdev](https://twitter.com/vectordotdev) on Twitter.
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:13]: Nit: in this list I think "Follow" etc works a bit better than "Following" etc. The Gerund Police are gonna let you off with a very light warning 👮🏼 THIS TIME.

File: docs/manual/about/release-process.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+* If you are using [one of our package
+	repositories](https://cloudsmith.io/~timber/repos/vector/packages/) you should
+	be able to see the update available when updating your package lists.
+
+## Release cadence
+
+We aim to:
+
+* Release a stable version of Vector every 6 weeks.
+* Release patch fixes as needed to fix high priority bugs and regressions.
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:28]: high-priority

File: docs/manual/about/release-process.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+version of Vector's version number. For example: `v0.11.1` to `v0.12.0`.
+
+### Patch releases
+
+Patch releases can be pushed out at any point to include regression and high
+priority bug fixes. We evaluate at the end of every two weeks whether a patch
+should be published for the current stable release, but can release outside of
+this schedule depending on need.
+
+By default, any bug fixes that are not regression fixes, will be left for the
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:49]: Remove second comma

File: docs/manual/about/release-process.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+next stable release.
+
+### Nightly releases
+
+We release a nightly version of Vector every day that represents the latest
+state of Vector. We aim to keep these as stable as possible, but they are
+inherently less stable than patch releases. We offer these to:
+
+* Allow users to beta test unreleased vector features and provide us with feedback
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:58]: s/vector/Vector

File: docs/manual/about/release-process.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+* Run blackbox integration tests against released artifacts
+
+These can be downloaded from via our [nightly downloads
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:61]: Make "these" more explicit, i.e. "These nightly releases"

Comment (lucperkins) [reply docs/manual/about/release-process.md:61]: Also remove either "from" or "via." I think either works just fine.

File: docs/manual/about/release-process.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+page](https://vector.dev/releases/nightly/download/) or via our our [nightly
+package
+repositories](https://cloudsmith.io/~timber/repos/vector-nightly/packages/).
+
+## Compatibility guarantees
+
+We aim to keep Vector as backwards compatible as possible, preferring
+deprecation over breaking compatibly, but we will occasionally introduce
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:69]: s/compatibly/compatibility

File: docs/manual/about/release-process.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+backwards incompatible changes as we learn better ways of doing things .
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:70]: backwards-incompatible

Comment (lucperkins) [reply docs/manual/about/release-process.md:70]: Also remove space before period

File: docs/manual/about/release-process.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+While Vector is pre-1.0, we will only make backwards incompatible changes in
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:72]: backwards-incompatible

(there are some other instances of this as well)

File: docs/manual/about/release-process.md (hunk 1 above)
Comment (lucperkins) [inline docs/manual/about/release-process.md:76]: "Has never happened yet" is a bit redundant. "has yet to happen" or "has never happened" flows a bit better methinks.

File: docs/manual/about/release-process.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+We always appreciate early feedback on Vector as we are developing it to help
+ensure the highest quality releases.
+
+If you are able to, running the [nightly
+release](https://vector.dev/releases/nightly/download/) of Vector allows you to
+test out unreleased features and fixes and provide feedback to guide our
+development. We aim to keep nightly as stable as possible through integration
+testing, but there will occasionally be issues that slip through and are fixed
+for the next nightly release. For example, you could run choose to run the
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:98]: "you could choose to run"

File: docs/manual/about/release-process.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+nightly version in your development environments and save stable for
+production.
+
+Otherwise, the stable release is your best bet.
```
Comment (lucperkins) [inline docs/manual/about/release-process.md:102]: This sentence should be appended to the previous paragraph

**Review by lucperkins — APPROVED** [review]
Just some non-blocking nits. LGTM!

**Comment by jszwedko** [comment]
Thanks for the thorough 👁️ @lucperkins ! I think I applied all of the suggestions.

File: docs/manual/about/release-process.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+	repositories](https://cloudsmith.io/~timber/repos/vector/packages/) you should
+	be able to see the update available when updating your package lists.
+
+## Release cadence
+
+We aim to:
+
+* Release a stable version of Vector every 6 weeks.
+* Release patch fixes as needed to fix high-priority bugs and regressions.
+* Release daily builds representing the latest state of Vector for feedback.
```
Comment (lukesteensen) [inline docs/manual/about/release-process.md:29]: Should we call these "nightly builds" instead of "daily builds" for consistency?

**Review by lukesteensen — APPROVED** [review]
Looks great, thanks @jszwedko!

**Review by binarylogic — APPROVED** [review]

File: docs/manual/about/release-process.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+	notified for new releases.
+* If you are using [one of our package
+	repositories](https://cloudsmith.io/~timber/repos/vector/packages/) you should
+	be able to see the update available when updating your package lists.
+
+## Release cadence
+
+We aim to:
+
+* Release a stable version of Vector every 6 weeks.
```
Comment (blt) [inline docs/manual/about/release-process.md:27]: I like the notion of a regular cadence.

**Review by blt — APPROVED** [review]

**Pull request merged**